### PR TITLE
Mdhwang/reduce mobile checks

### PR DESCRIPTION
**Pull request by mdh1418**
This is a follow up to PR #34662. 
This PR simplifies conditional checks `'$(TargetOS)' == 'iOS' or '$(TargetOS)' == 'Android' or '$(TargetOS)' == 'tvOS'` and all other permutations with `'$(TargetsMobile)' == 'true'`
#35023

### CODE REVIEW

File: eng/Subsets.props [hunk 1]
```diff
@@ -35,10 +35,13 @@
       You can get around this limitation by running the corehost subset, manually copying host
       artifacts to the test layout, then running the test subset.
   -->
+  <PropertyGroup>
+    <TargetsMobile Condition="'$(TargetOS)' == 'iOS' or '$(TargetOS)' == 'Android' or '$(TargetOS)' == 'tvOS'">true</TargetsMobile>
```
Comment (ViktorHofer) [inline eng/Subsets.props:39]: This should be moved to https://github.com/dotnet/runtime/blob/master/eng/Configurations.props

Comment (mdh1418) [reply eng/Subsets.props]: Thanks!

**Review by ViktorHofer — APPROVED** [review]

File: src/installer/pkg/projects/Directory.Build.targets [hunk 2]
```diff
@@ -19,7 +19,7 @@
     out and skip building. 
   -->
   <Target Name="MobileSkipBuildProps" 
-          Condition="'$(SkipBuildOnRuntimePackOnlyOS)' == 'true' and ('$(TargetOS)' == 'iOS' or '$(TargetOS)' == 'tvOS' or '$(TargetOS)' == 'Android')"
+          Condition="'$(SkipBuildOnRuntimePackOnlyOS)' == 'true' and '$(TargetsMobile)' == 'true'"
```
Comment (safern) [inline src/installer/pkg/projects/Directory.Build.targets:22]: Leave this one as is, this is taking care here: https://github.com/dotnet/runtime/pull/34980/files#diff-0a57cd77a878d620b1edc73b4dcc9f1fR22

File: src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Runtime.pkgproj [hunk 3]
```diff
@@ -5,9 +5,7 @@
     shared framework on those platforms.
   -->
   <PropertyGroup>
-    <ExcludeHostAssets Condition="'$(TargetOS)' == 'iOS'">true</ExcludeHostAssets>
-    <ExcludeHostAssets Condition="'$(TargetOS)' == 'tvOS'">true</ExcludeHostAssets>
-    <ExcludeHostAssets Condition="'$(TargetOS)' == 'Android'">true</ExcludeHostAssets>
+    <ExcludeHostAssets Condition="'$(TargetsMobile)' == 'true'">true</ExcludeHostAssets>
```
Comment (safern) [inline src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Runtime.pkgproj:8]: This is also taken care of here: https://github.com/dotnet/runtime/pull/34980/files#diff-a956a8da71efa9dfd68d21cab8f557e5R8

File: src/libraries/Directory.Build.props [hunk 4]
```diff
@@ -35,17 +35,13 @@
...
+    <TargetArchitecture Condition="'$(TargetArchitecture)' == '' and '$(TargetsMobile)'">x64</TargetArchitecture>
     <TargetArchitecture Condition="'$(TargetArchitecture)' == ''">x64</TargetArchitecture>
 
     <!-- RuntimeOS is calculated based on the build system OS, however if building for WebAssembly/iOS/Android we need to let
          the build system to use webassembly/ios/android as the RuntimeOS for produced package RIDs. -->
     <RuntimeOS Condition="'$(TargetOS)' == 'WebAssembly'">$(TargetOS.ToLowerInvariant())</RuntimeOS>
-    <RuntimeOS Condition="'$(TargetOS)' == 'iOS'">$(TargetOS.ToLowerInvariant())</RuntimeOS>
-    <RuntimeOS Condition="'$(TargetOS)' == 'tvOS'">$(TargetOS.ToLowerInvariant())</RuntimeOS>
-    <RuntimeOS Condition="'$(TargetOS)' == 'Android'">$(TargetOS.ToLowerInvariant())</RuntimeOS>
+    <RuntimeOS Condition="'$(TargetsMobile)' == 'true'">$(TargetOS.ToLowerInvariant())</RuntimeOS>
```
Comment (safern) [inline src/libraries/Directory.Build.props:44]: Nit: Maybe we can merge this with the `WebAssembly` definition.

**Comment by safern** [comment]
If PR: #34980 is merged first, maybe consider removing these: https://github.com/dotnet/runtime/pull/34980/files#diff-9f3c793773fc53c31ea09d056e98a4eeR200

as I believe `eng/Configurations.props` is imported for all MSBuild paths early.

File: src/libraries/Directory.Build.props (hunk 4 above)
Comment (mdh1418) [reply src/libraries/Directory.Build.props]: Done

File: src/installer/pkg/projects/Directory.Build.targets (hunk 2 above)
Comment (mdh1418) [reply src/installer/pkg/projects/Directory.Build.targets]: Thanks, fixed merge conflicts

File: src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Runtime.pkgproj (hunk 3 above)
Comment (mdh1418) [reply src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Runtime.pkgproj]: 👍

File: src/libraries/Directory.Build.props [hunk 5]
```diff
@@ -35,17 +35,12 @@
     <TargetArchitecture Condition="'$(TargetArchitecture)' == '' and '$(HostArch)' == 'arm'">arm</TargetArchitecture>
     <TargetArchitecture Condition="'$(TargetArchitecture)' == '' and '$(HostArch)' == 'arm64'">arm64</TargetArchitecture>
     <TargetArchitecture Condition="'$(TargetArchitecture)' == '' and '$(TargetOS)' == 'WebAssembly'">wasm</TargetArchitecture>
-    <TargetArchitecture Condition="'$(TargetArchitecture)' == '' and '$(TargetOS)' == 'iOS'">x64</TargetArchitecture>
-    <TargetArchitecture Condition="'$(TargetArchitecture)' == '' and '$(TargetOS)' == 'tvOS'">x64</TargetArchitecture>
-    <TargetArchitecture Condition="'$(TargetArchitecture)' == '' and '$(TargetOS)' == 'Android'">x64</TargetArchitecture>
+    <TargetArchitecture Condition="'$(TargetArchitecture)' == '' and '$(TargetsMobile)'">x64</TargetArchitecture>
```
Comment (mdh1418) [inline src/libraries/Directory.Build.props:38]: Forgot to add `== 'true'`. Testing locally to see if it builds the testhost folder in `artifacts/bin/`

**Pull request merged**